src/helpers/twitter/image-helper.php

```diff
@@ -44,13 +44,15 @@ public function get_image_size() {
 	/**
 	 * Retrieves an image url by its id.
 	 *
-	 * @codeCoverageIgnore It is a wrapper method.
-	 *
 	 * @param int $image_id The image id.
 	 *
-	 * @return string The image url.
+	 * @return string The image url. Empty string if the attachment is not valid.
 	 */
 	public function get_by_id( $image_id ) {
+		if ( ! $this->image->is_valid_attachment( $image_id ) ) {
+			return '';
+		}
```

Comment:
Is this safe to change? Maybe you should add a check where it is used?
- https://github.com/Yoast/wordpress-seo/blob/trunk/src/builders/indexable-social-image-trait.php#L137
- https://github.com/Yoast/wordpress-seo/blob/53bc87439e18e7142971446d19c6accaad9e8242/src/builders/indexable-social-image-trait.php#L78
Currently those default to `NULL` instead of an empty string

Reply:
I just see on [line 132](https://github.com/Yoast/wordpress-seo/blob/trunk/src/builders/indexable-social-image-trait.php#L132) we check `if ( $indexable->twitter_image ...`. Both `null` and `''` evaluate to false. Same goes for [here](https://github.com/Yoast/wordpress-seo/blob/2394f62572096e62aefc603ffab04a8e57f8deb9/src/generators/twitter-image-generator.php#L82-L83).

Reply:
Thinking about this some more, it is not the same value. You are also storing a string in the database, which for the indexables is set up to have `NULL` as no value. Let's not pollute that with `NULL` or empty string?

What is your reasoning for wanting an empty string?

Reply:
Yeah, the reasoning was preserving as much as possible the original behaviour: in this case, the `get_by_id` method was returning directly whatever comes from `Image_Helper::get_attachment_image_source` which, in case no attachment is found, [would return empty string](https://github.com/Yoast/wordpress-seo/blob/trunk/src/helpers/image-helper.php#L127C1-L134C21).